ktor-network/ktor-network-tls/jvm/test/io/ktor/network/tls/tests/ConnectionTests.kt

```diff
@@ -66,4 +79,70 @@ class ConnectionTests {
         output.close()
         socket.close()
     }
+
+    @Test
+    fun clientCertificatesAuthTest() {
+        val keyStoreFile = File("build/temp.jks")
+        val keyStore = generateCertificate(keyStoreFile, algorithm = "SHA256withRSA", keySizeInBits = 4096)
+        val chain1 = keyStore.getCertificateChain("mykey").toList() as List<X509Certificate>
+        val certs = chain1.toList().toTypedArray()
+        val password = "changeit".toCharArray()
+        val pk = keyStore.getKey("mykey", password) as PrivateKey
+        val trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm()) .also { it.init(keyStore) }
+        val workerGroup: EventLoopGroup = NioEventLoopGroup()
+        try {
+            val b = ServerBootstrap()
```

Comment:
One last little note. Could you avoid using shortcuts and single letters in variable names?

Reply:
Thanks. I am sorry, not sure what `shortcuts` you are referring to, could you please elaborate? Variable names fixed.

Reply:
that's what I mean. thanks, lgtm